packages/optimizers/esbuild/src/ESBuildOptimizer.js

```diff
@@ -0,0 +1,64 @@
+// @flow
+
+import {startService} from 'esbuild';
+import {Optimizer} from '@parcel/plugin';
+import {blobToString, normalizePath} from '@parcel/utils';
+import SourceMap from '@parcel/source-map';
+import path from 'path';
+
+let service = null;
+export default (new Optimizer({
+  async optimize({contents, map, bundle, options, getSourceMapReference}) {
+    if (!bundle.env.minify) {
+      return {contents, map};
+    }
+
+    if (!service) {
+      service = await startService();
+    }
+
+    let relativeBundlePath = path.relative(
+      options.projectRoot,
+      bundle.filePath,
+    );
+    let code = await blobToString(contents);
+    if (map && options.sourceMaps) {
+      let vlqMappings = map.toVLQ();
+      vlqMappings = {
```

Comment:
Why not use `map.stringify`?

Reply:
mainly performance, there's some overhead for using async and the additional logic in stringify. Probably fairly minimal, so we could use it the optimizers don't run that often anyway